### PR TITLE
feat: create a man page

### PR DESCRIPTION
**Pull request by TibebeJS**
- added a `man` page generator `npm` script
- added a `man` support on `package.json`
- added a sample man page entry

man-page-screenshot

### CODE REVIEW

**Comment by TibebeJS** [comment]
closes #121

File: package.json [hunk 1]
```diff
@@ -10,8 +10,10 @@
     "test": "echo \"Test will be implemented.\" && exit 0",
     "start": "./bin/gtop",
     "lint": "prettier --single-quote --trailing-comma es5 --write 'lib/**/*.js'",
-    "lint-check": "prettier --check --single-quote --trailing-comma es5 'lib/**/*.js'"
+    "lint-check": "prettier --check --single-quote --trailing-comma es5 'lib/**/*.js'",
+    "man": "cd man && marked-man --version 'v1.1.2' --manual 'Gtop Help' --section 1 man.md > gtop.1"
```
Comment (aksakalli) [inline package.json:14]: is it possible to not to hardcode the version in here?

File: package.json [hunk 2]
```diff
@@ -10,8 +10,10 @@
...
   },
+  "man": ["man/gtop.1"],
```
Comment (aksakalli) [inline package.json:16]: I am not familiar with man in `package.json`. That would add the man entry after `npm i -g gtop`?

**Review by aksakalli — COMMENTED** [review]
Hi :) Thank you for your contribution. I have some questions.

File: package.json (hunk 1 above)
Comment (TibebeJS) [reply package.json]: yes, it is possible. I've pushed a commit that address that.

File: package.json (hunk 2 above)
Comment (TibebeJS) [reply package.json:16]: yes, that's correct

Comment (TibebeJS) [reply package.json:16]: Here is an excerpt from the official documentation.
https://docs.npmjs.com/cli/v8/configuring-npm/package-json#man

> **man**
>
> Specify either a single file or an array of filenames to put in place for the man program to find.
> 
> If only a single file is provided, then it's installed such that it is the result from man <pkgname>, regardless of its actual filename.

**Comment by TibebeJS** [comment]
I have removed the separate `man` npm script for the roff generator since it requires a manual execution of `npm run man` for it to work and opted to use `prepublishOnly` instead.
This ensures that an updated `man` page will be generated everytime the package is `publish`ed. (Without having to edit the CI pipeline)

**Comment by aksakalli** [comment]
> @aksakalli I have removed the separate `man` npm script for the roff generator since it requires a manual execution of `npm run man` for it to work and opted to use `prepublishOnly` instead. This ensures that an updated `man` page will be generated everytime the package is `publish`ed. (Without having to edit the CI pipeline)

That sounds great. I was worried about adding extra manual steps to releasing a new version. I guess `man/gtop.1` just becomes a placeholder. Is it necessary to commit it to git?

**Comment by aksakalli** [comment]
> > 
> 
> The `gtop.1` file is auto-generated from the `md` file and since we do track the `.md` file's revisions, it's not really necessary to commit `gtop.1` I guess (but it wouldn't hurt :) )

I would say it's better to remove `gtop.1` from the source. I found a good repo as an example: [gulp-cli](https://github.com/gulpjs/gulp-cli). We can do the following changes:

* remove `./man/gtop.1`
* move `./man/cli.md` to `./CLI.md` (that could be useful as a CLI documentation page as well in the root.)
* In `package.json`
  * `"man": "gtop.1"`
  * `"prepublish": "marked-man --name gtop ./CLI.md > gtop.1"`

**Comment by TibebeJS** [comment]
> * remove `./man/gtop.1`
> * move `./man/cli.md` to `./CLI.md` (that could be useful as a CLI documentation page as well in the root.)

Should we dedicate a separate `/docs` folder like `gulp-cli` (incase there arises a need for more documentation in the future)?

**Comment by aksakalli** [comment]
> > * remove `./man/gtop.1`
> > * move `./man/cli.md` to `./CLI.md` (that could be useful as a CLI documentation page as well in the root.)
> 
> Should we dedicate a separate `/docs` folder like `gulp-cli` (incase there arises a need for more documentation in the future)?

Not for now, we can create a folder in future if needed.

**Comment by TibebeJS** [comment]
@aksakalli I've pushed changes. But here couple of things to note:

> `"prepublish": "marked-man --name gtop ./CLI.md > gtop.1"`

- [`prepublish` has been splitted into 2 commands since node 4.x ](https://iamakulov.com/notes/npm-4-prepublish/#whats-next) (`prepublishOnly` and `prepare`) 

> `"man": "gtop.1"` 

There are [known issues when `man` field is just a string](https://github.com/npm/bin-links/issues/25) (even-though the documentation says otherwise)


So I have taken those 2 points into account in my commit. :)

**Comment by aksakalli** [comment]
Thank you for your contribution. ❤️

**Pull request merged**